rust/ql/lib/codeql/rust/controlflow/BasicBlocks.qll

```diff
@@ -0,0 +1,262 @@
+private import rust
+private import ControlFlowGraph
+private import internal.SuccessorType
+private import SuccessorTypes
+private import internal.ControlFlowGraphImpl as Impl
+private import codeql.rust.generated.Raw
+private import codeql.rust.generated.Synth
+
+/**
+ * A basic block, that is, a maximal straight-line sequence of control flow nodes
+ * without branches or joins.
+ */
+class BasicBlock extends TBasicBlockStart {
```

Comment:
final

Reply:
`BasicBlock` is extended further down in the file so I did
```ql
final class BasicBlock = BasicBlockImpl;

private class BasicBlockImpl extends TBasicBlockStart { ... }
```
similar to what you suggested for `SuccessorType`.